torch/distributed/distributed_c10d.py

```diff
@@ -1700,7 +1700,7 @@ def gather_object(obj, object_gather_list=None, dst=0, group=None):
         object_gather_list[i] = _tensor_to_object(tensor, tensor_size)
 
 
-def broadcast_object_list(object_list, src=0, group=None):
+def broadcast_object_list(object_list, src=0, group=None, dist_device=None):
```

Comment:
shall we add test for this new argument?

Reply:
Added one test in test_broadcast_object_list